NickvisionCavalier.GNOME/Controls/MessageDialog.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+
+namespace NickvisionCavalier.GNOME.Controls;
+
+/// <summary>
+/// Responses for the MessageDialog
+/// </summary>
+public enum MessageDialogResponse
+{
+    Suggested,
+    Destructive,
+    Cancel
+}
+
+/// <summary>
+/// A dialog for showing a message
+/// </summary>
+public class MessageDialog
+{
+    private readonly Adw.MessageDialog _dialog;
+    
+    public event EventHandler<MessageDialogResponse>? OnDialogClosed;
```

Comment:
Don't need...have `OnResponse` handle....see Denaro/Parabolic/Tagger's use of MessageDialog...

Reply:
I started with MessageDialog from Denaro/Parabolic and changed it because I find using `g_main_context_iteration` kinda hacky 😄

Reply:
Where do we use `g_main_context_iteration`? Do you mean `g_main_context_invoke` where we have to save the callbacks? The won't change with making you own EventHandler because the gir.core EventHandler is implemented the same way. 

The problem is that gir.core 0.3.0 (0.4.0 will fix this) doesn't have a SynchronizationContext to restore the thread when using async methods, so callbacks need to be saved after an async method either way. If none of the code is async, then saving callbacks isn't needed.

Reply:
No, I mean `g_context_iteration`. In other apps MessageDialog has `Run` method, that is called to show dialog and then  there is `while (dialog.Visible) { g_main_context_iteration ...` and in the end this method returns response.